template.rb

```diff
@@ -167,6 +168,41 @@ def setup_timezone
   gsub_file("config/application.rb", "Central Time (US & Canada)", "Adelaide")
 end
 
+# By default in production Rails is using TaggedLogging
+# replace it with StructuredLogging(with json formatter)
+# and add StructuredLogging(with basic formatter) to development and test
+# Note: should be invoked before #setup_staging
+def setup_logger
+  copy_file("lib/loggers/structured_logging.rb")
+
+  insert_into_file("config/application.rb", after: "Bundler.require(*Rails.groups)\n") do
+    "require_relative \"../lib/loggers/structured_logging\"\n"
+  end
+
+  gsub_file("config/environments/production.rb", "::Logger::Formatter", "::StructuredLogging::JsonFormatter")
+  gsub_file("config/environments/production.rb", "ActiveSupport::TaggedLogging.new(logger)",
+            "::StructuredLogging.new(logger)")
+
+  insert_into_file("config/environments/production.rb", before: "config.logger") do
+    "STDOUT.sync = true\n  "
```

Comment:
why is there whitespace after the \n?
you could add the whitespace to the pattern (before) if you're seeing the code inserted in the middle of the line

Reply:
This is to align the config block with the correct spacing..